### ISSUE TITLE
Maybe a issue with SetPropIndex (2612)?

### ISSUE THREAD

**Issue by ghost** [citizenfx/fivem#1486]
**Description:**

When using the build `2612` and adding more add-on clothes props, the 'hats' don't seem to set the model properly; I don't quite know if it's not setting the model properly, or if it's just completely broken, but what I do know is, that it just doesn't show up at all.

**Example:** If you have added any add-on 'hats', and swap to where it is supposed to be located, *(in my case it would be slot # 214 in the 'hats' section)* it doesn't show up. But then when you downgrade to build `2189` (Perico), all the hats show up properly. This only seems to be happening for 'hats', everything else is showing up / working properly..

**Note:** I was able to reproduce this *every single time* upon switching builds; *every time* I swapped to `2612`, all of the new add-on props under the 'hats' section were nowhere to be found. There wasn't *any* resources changed, and if you can't tell from the videos below, I am using 'fivem-appearance'.

**Edit:** **I have also came across someone else reporting this issue, you can find the topic here. --> [Click me to view the topic](https://forum.cfx.re/t/helmets-hats-disappearing/4864117)**

**Expected Behavior:**

- The ability to use add-on hats. *(for some reason everything else works)*

**Videos:**
- [Streamable #1 (2189)](https://streamable.com/toibvk)
- [Streamable #2 (2612)](https://streamable.com/i9akdv)

**Additional Context**
- **Game Version:** 5562 (Canary)
- **Server Version:** 5644
- **Build:** 2189 / 2612

**Comment by PoorLatency** [comment]
This is still an issue the game versions; **2612** and **2666** (tested on 5824 server version)., the addon hats don't appear it seems as Rockstar changed something when you have a lot of addon hats. Hats work fine on other builds **2189**, etc.

**Comment by vicegamingnetwork** [comment]
https://github.com/DurtyFree/durty-cloth-tool/wiki/YMT-game-limit-and-crash-issues

While this post speaks of additional ymt file slots, if you lower your .ymt files to 6... Everything will stream appropriately.

**Comment by Moozdzn** [comment]
@vicegamingnetwork Not related. Repro in [Helmets/Hats Disappearing](https://forum.cfx.re/t/helmets-hats-disappearing/4864117) only uses 2 .ymt files. 
Also, going over the Game build limit produces a crash and not ped props not appearing.

**Comment by gottfriedleibniz** [comment]
[Unfortunately](https://forum.cfx.re/t/help-build-2372-crash-to-desktop/4123759/13) this issue is a bit difficult to deal with. Closing since this issue hasn't been updated since 2022 - the forums contain all relevant conversations and information.

**Comment by Moozdzn** [comment]
@gottfriedleibniz This issue is about some other prop limit, unrelated to ymt limits. Specifically props such as hats not spawning after a certain amount is being streamed (dlc dependent). This is my repo https://forum.cfx.re/t/helmets-hats-disappearing/4864117 and im only using 2 ymt. I do understand if anything clothing related is hard to fix and thats the reason this was closed.

**Comment by grzybeek** [comment]
Just to provide additional context and an explanation for why this is happening:

I believe that some ped prop stuff in the code is uint8, meaning there is a "global limit" of 255 maximum props of one type. If you exceed this and add more props, game will render only number of props that you are above limit (for example, if you add 260 p_head props, you will be able to select only first 5 props, anything else will not load at all)

Of course R* adds more and more props with each DLC, and this is why users are reporting something works on one gamebuild but doesn't on newer ones.

I hope this somehow helps to anyone trying to solve this issue :)

**Comment by gottfriedleibniz** [comment]
@grzybeek, is this documented anywhere?

**Comment by grzybeek** [comment]
@gottfriedleibniz I don't think so, just know this from my own experience with peds/clothing 😄

**Comment by manzarek123** [comment]
Yes there is a 255 limit, a repro was provided here at disquse's request for investigation
https://forum.cfx.re/t/request-to-increase-head-prop-limit-in-fivem/5201885/8

With every new dlc the amount of hat that can be streamed is less and less :(

**Comment by gottfriedleibniz** [comment]
It would be quite nice to have a place that centralizes all of the game limits everyone here has run into and/or are fighting against.

**Comment by Twrntyways** [comment]
I've tried 2802/2944 and 2699 and I've gone from just one add on hat to a max of 10 none show and only a limited amount of gta hats show in 2954 it 46 or so in 2802 it's like 26 and so on the rest are blank with a -1 as the texture. It's only hats and I inky have one male pack 2 ymts and one female pack. I wish someone would direct us in the right direction. I've had devs and even the pack creator want to charge me to tell me the fix. This is crazy

**Comment by Legacy-TacticalGamingInteractive** [comment]
> It would be quite nice to have a place that centralizes all of the game limits everyone here has run into and/or are fighting against.

https://github.com/DurtyFree/durty-cloth-tool/wiki/YMT-game-limit-and-crash-issues

I also wanted to post this for you to check. Thats what most of us are going off of as well as its been researched quite a bit by the community.

For our server we experience the 255 limit then we get the invisible props as mentioned. Tested on both 2699 and 3095. Hopefully this workaround can help out with that.

It reminds me of the 255 siren ID's limit where someone figured out how to fix that for client sided only. I can't speak for this one though as I've not see anyone fix it like that yet.
